muikku-school-data-pyramus/src/main/java/fi/otavanopisto/muikku/plugins/schooldatapyramus/PyramusUpdater.java

```diff
@@ -628,6 +628,15 @@ public boolean updateCourseStudent(Long courseStudentId, Long courseId, Long stu
     
     return false;
   }
+  
+  public void removeCourseStudent(Long courseStudentId, Long courseId, Long studentId) {
+    String identifier = identifierMapper.getWorkspaceStudentIdentifier(courseStudentId);
+    SchoolDataIdentifier workspaceUserIdentifier = new SchoolDataIdentifier(identifier, SchoolDataPyramusPluginDescriptor.SCHOOL_DATA_SOURCE);
+    WorkspaceUserEntity workspaceUserEntity = workspaceUserEntityController.findWorkspaceUserEntityByWorkspaceUserIdentifierIncludeArchived(workspaceUserIdentifier);
+    if (workspaceUserEntity != null) {
+      workspaceUserEntityController.deleteWorkspaceUserEntity(workspaceUserEntity);
```

Comment:
Admittedly not 100% safe. References WorkspaceUserPermissionOverride, which has never been used so far, and AssessmentRequestMessageId, which is used in code but seems to serve no actual reason in the application itself (that, or the functionality planned to make use of that information has not yet been implemented, whatever and whenever that would be).